File: bindings/dotnet/examples/Df.cs
using System;
using Hyperic.Sigar;

public class Df {

    private static String FormatSize(long value) {
        return Sigar.FormatSize(value * 1024);
    }

    public static void Main() {
        Sigar sigar = new Sigar();

        foreach (FileSystem fs in sigar.FileSystemList()) {
            FileSystemUsage usage;
            long used, avail, total, pct;

            try {
                usage = sigar.FileSystemUsage(fs.DirName);

                used = usage.Total - usage.Free;
                avail = usage.Avail;
                total = usage.Total;
                pct = (long)(usage.UsePercent * 100);
            } catch (SigarException) {
                used = avail = total = pct = 0;
                continue;
            }

            string usePct;
            if (pct == 0) {
                usePct = "-";
            }
            else {
                usePct = pct + "%";
            }

            System.Console.WriteLine(fs.DevName + "\t" +
                                     FormatSize(total) + "\t" +
                                     FormatSize(used) + "\t" + 
                                     FormatSize(avail) + "\t" + 
                                     usePct + "\t" +
                                     fs.DirName + "\t" + 
                                     fs.SysTypeName + "/" + fs.TypeName);
        }
    }
}

File: bindings/ruby/examples/df.rb
require 'rbsigar'

def format_size(size)
  return Sigar.format_size(size * 1024)
end

sigar = Sigar.new
fslist = sigar.file_system_list

puts "Filesystem\tSize\tUsed\tAvail\tUse%\tMounted on\tType\n"

fslist.each do |fs|
  dir_name = fs.dir_name
  usage = sigar.file_system_usage(dir_name)

  total = usage.total
  used = total - usage.free
  avail = usage.avail
  pct = usage.use_percent * 100

  puts fs.dev_name + "\t" +
    format_size(total) + "\t" +
    format_size(used) + "\t" +
    format_size(avail) + "\t" +
    (pct == 0.0 ? '-' : pct.to_s) + "\t" +
    dir_name + "\t" + 
    fs.sys_type_name + "/" + fs.type_name
end
